### DISCUSSION TITLE
How do I create empty chart and update with realtime data

### DISCUSSION THREAD

**Discussion by armeya25** [highfestiva/finplot#443] [Q&A] [answered]
I was trying to create empty plot and after that update it with realtime data.

Just want to plot it no any indicator or anything just simple candlestick chart only.

How do I do it.
Example provided are to complex to understand,
I am dumb

**Comment by highfestiva** [comment, marked as answer]
There are two examples included which updates in real time. One uses http calls to fetch new data, and stalls the whole plot during that time (bfx.py), and the other uses websockets and is a bit more involved (complicated.py).  finplot is for plotting, not for fetching data, so you'll have to build that part yourself. Best is perhaps to use `complicated.py` and strip away code you don't need and you'll end up with a few lines that you can modify to suit your needs.

**Reply by armeya25** [reply]
Thank you for your fast reply

**Reply by armeya25** [reply]
I did write a function which reads local CSV file and return data line by line,
I was trying to add that data as realtime data and plot it

**Reply by armeya25** [reply]
import polars as pl
import pandas as pd
from time import sleep

class Data_Streamer:
    def __init__(self) -> None:
        self.pl_df = pl.DataFrame()
        self.pd_df = pd.DataFrame()

    def Csv_Streamer(self, file_name, to_pandas = False, sleep_time = 0.1) -> pl.DataFrame:
        """
        params : file_name  -> csv file name
        params : to_pandas  -> 
                            dafault false -> retruns polars datframe \n
                            if set to True -> return pandas datframe
        params : sleep_time -> return values after this much time
                            -> default value is 0.1 
        """
        df = pl.read_csv(file_name)
        length = df.height

        ## read csv file one row at a time
        reader =  pl.read_csv_batched(file_name, batch_size=1)
        if to_pandas is True:
            for i in range(length):
                sleep(sleep_time)
                values = reader.next_batches(n=1)
                for vals in values:
                    ## return pandas datframe
                    yield vals.to_pandas()
        else:
            for i in range(length):
                sleep(sleep_time)
                values = reader.next_batches(n=1)
                for vals in values:
                    ## return polars datframe
                    yield vals

**Reply by armeya25** [reply]
It streams data just like realtime data,
Want to plot it

**Reply by highfestiva** [reply]
If you post a complete, minimal working example, I can help you get it running.

**Reply by armeya25** [reply]
[testDF.csv](https://github.com/highfestiva/finplot/files/12370383/testDF.csv)
[check_me.txt](https://github.com/highfestiva/finplot/files/12370429/check_me.txt)

**Reply by highfestiva** [reply]
Here's a start:

```python
import finplot as fplt
import pandas as pd

def get_plot_update():
    df = pd.read_csv('testDF.csv')
    df['date'] = pd.to_datetime(df.date)
    df = df.set_index('date')
    for i in range(50, len(df)):
        yield df.iloc[0:i]

def animate(gen):
    df = next(gen)
    print(df)
    live_plot.candlestick_ochl(df['open close high low'.split()])

fplt.create_plot()
live_plot = fplt.live()
gen = get_plot_update()
fplt.timer_callback(lambda: animate(gen), 1)
fplt.show()
```

**Reply by armeya25** [reply]
works like charm
